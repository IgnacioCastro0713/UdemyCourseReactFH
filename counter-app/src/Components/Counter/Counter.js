import React, { useState } from 'react';
import PropTypes from 'prop-types';

function Counter({ value }) {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);
  const handleRemove = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <div>
      <h1>Counter App</h1>
      <h2>{ counter }</h2>
      <button onClick={ handleAdd }> +</button>
      <button onClick={ handleRemove }> -</button>
      <button onClick={ handleReset }> reset</button>
    </div>
  );
}

Counter.propTypes = {
  value: PropTypes.number
};

Counter.defaultProps = {
  value: 10
};

export default Counter;