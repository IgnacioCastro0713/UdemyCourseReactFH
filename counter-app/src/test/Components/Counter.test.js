import React from "react";
import { shallow } from "enzyme";
import Counter from "../../Components/Counter";


describe('Tests <Counter />', () => {

  let wrapper = shallow(<Counter/>);

  beforeEach(() => wrapper = shallow(<Counter/>));

  test('should show <Counter /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show the default value of 100', () => {
    const wrapper = shallow(<Counter value={ 100 }/>);
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('100');
  });

  test('must increase with the +1 button', () => {
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('11');
  });

  test('must be decremented with the -1 button', () => {
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('9');
  });

  test('must set the default value with the reset btn', () => {
    const wrapper = shallow(<Counter value={ 105 }/>);

    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(2).simulate('click');

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('105');
  });

});
