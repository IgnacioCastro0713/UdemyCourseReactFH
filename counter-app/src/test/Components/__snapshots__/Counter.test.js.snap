// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <Counter /> debe de mostrar <Counter /> correctamente 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
     +
  </button>
  <button
    onClick={[Function]}
  >
     -
  </button>
  <button
    onClick={[Function]}
  >
     reset
  </button>
</div>
`;

exports[`Pruebas en el <Counter /> should show <Counter /> correctly 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
     +
  </button>
  <button
    onClick={[Function]}
  >
     -
  </button>
  <button
    onClick={[Function]}
  >
     reset
  </button>
</div>
`;

exports[`Tests <Counter /> should show <Counter /> correctly 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
     +
  </button>
  <button
    onClick={[Function]}
  >
     -
  </button>
  <button
    onClick={[Function]}
  >
     reset
  </button>
</div>
`;
